shinylive/_main.py

```diff
@@ -469,6 +471,153 @@ def defunct_help(cmd: str) -> str:
+    required=True,
+    default="editor",
+    help="The shinylive mode: include the editor or show only the app.",
+)
+@click.option(
+    "-l",
+    "--language",
+    type=click.Choice(["python", "py", "R", "r"]),
+    required=False,
+    default=None,
+    help="The primary language used to run the app, by default inferred from the app file.",
+)
+@click.option(
+    "-v", "--view", is_flag=True, default=False, help="Open the link in a browser."
+)
+@click.option(
+    "--no-header", is_flag=True, default=False, help="Hide the Shinylive header."
+)
+@click.argument("app", type=str, nargs=1, required=True, default="-")
+@click.argument("files", type=str, nargs=-1, required=False)
+def encode(
+    app: str,
+    files: Optional[tuple[str, ...]] = None,
+    mode: str = "editor",
+    language: Optional[str] = None,
+    no_header: bool = False,
+    view: bool = False,
+) -> None:
+    if app == "-":
+        app_in = sys.stdin.read()
+    else:
+        app_in = app
+
+    url = encode_shinylive_url(
+        app=app_in,
+        files=files,
+        mode=mode,
+        language=language,
+        header=not no_header,
+    )
+
+    print(url)
+
+    if view:
+        import webbrowser
+
+        webbrowser.open(url)
+    return
+
+
+@url.command(
+    short_help="Decode a shinylive.io URL.",
+    help="""
+Decode a shinylive.io URL.
+
+URL is the shinylive editor or app URL. If not specified, the URL will be read from
+stdin, allowing you to read the URL from a file or the clipboard.
+
+On macOS, you can read the URL from the clipboard with:
+
+    pbpaste | shinylive url decode
+""",
+)
+@click.option(
+    "--dir",
+    type=str,
+    default=None,
+    help="Output directory into which the app's files will be written. The directory is created if it does not exist. ",
+)
+@click.option(
+    "--json",
+    is_flag=True,
+    default=False,
+    help="Prints the decoded shinylive bundle as JSON to stdout, ignoring --out.",
+)
+@click.argument("url", type=str, nargs=1, default="-")
+def decode(url: str, dir: Optional[str] = None, json: bool = False) -> None:
+    if url == "-":
+        url_in = sys.stdin.read()
+    else:
+        url_in = url
+    bundle = decode_shinylive_url(str(url_in))
+
+    if json:
+        print_as_json(bundle)
+        return
+
+    if dir is not None:
+        import os
+
+        out_dir = Path(dir)
+        os.makedirs(out_dir, exist_ok=True)
+        for file in bundle:
+            with open(out_dir / file["name"], "w") as f_out:
+                f_out.write(
+                    file["content"].encode("utf-8", errors="ignore").decode("utf-8")
+                )
+    else:
+        print("")
+        for file in bundle:
+            print(f"## file: {file['name']}")
+            if "type" in file and file["type"] == "binary":
+                print("## type: binary")
+            print("")
+            print(file["content"].encode("utf-8", errors="ignore").decode("utf-8"))
```

Comment:
I just want to make sure: for binary files, this will be base64-encoded content, right? (Because it would have been base-64 encoded before being serialized to JSON and then encoded with lzstring.)

Reply:
Yeah that's right -- this case was correct, since the `shinylive-python` or `shinylive-r` chunk includes the base64-encoded content.